From bb608b4115c6d1f0741c8863616b36d50cb6e1ec Mon Sep 17 00:00:00 2001
From: Shane Synan <1449717+digitalcircuit@users.noreply.github.com>
Date: Mon, 25 Jan 2021 19:07:13 -0500
Subject: [PATCH] settings: Fix ignoring 'securecore: false' default

Don't use || to set default value for "securecore", instead check
the type of the variable.  Otherwise, false evaluates to, well, false,
and "securecore" can never be defaulted to false.  This fixes ignoring
the custom default setting "securecore: false" in settings-user.json
---
 public/javascripts/angular-init.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/public/javascripts/angular-init.js b/public/javascripts/angular-init.js
index 1c8fb2d..ee2df6c 100644
--- a/public/javascripts/angular-init.js
+++ b/public/javascripts/angular-init.js
@@ -157,7 +157,7 @@ angular.module('quassel', ['ngQuassel', 'ngAria', 'ngSanitize', 'ui.bootstrap',
             set('port', data.settings.port);
             set('initialBacklogLimit', data.settings.initialBacklogLimit);
             set('backlogLimit', data.settings.backlogLimit);
-            set('securecore', data.settings.securecore || true);
+            set('securecore', (typeof data.settings.securecore === 'boolean' ? data.settings.securecore : true));
             set('theme', data.settings.theme);
             set('perchathistory', data.settings.perchathistory);
             set('displayfullhostmask', data.settings.displayfullhostmask);
